Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/RagdollColliderTranslationManipulators.cpp

```diff
@@ -0,0 +1,193 @@
+
+        m_translationManipulators.InstallPlanarManipulatorMouseMoveCallback(
+            [this](const AzToolsFramework::PlanarManipulator::Action& action)
+            {
+                OnManipulatorMoved(action.m_start.m_localPosition, action.m_current.m_localOffset);
+            });
+
+        m_translationManipulators.InstallSurfaceManipulatorMouseMoveCallback(
+            [this](const AzToolsFramework::SurfaceManipulator::Action& action)
+            {
+                OnManipulatorMoved(action.m_start.m_localPosition, action.m_current.m_localOffset);
+            });
+
+        // mouse up callbacks
+        m_translationManipulators.InstallLinearManipulatorMouseUpCallback(
+            [this](const AzToolsFramework::LinearManipulator::Action& action)
+            {
+                FinishEditing(action.m_start.m_localPosition, action.m_current.m_localPositionOffset);
+            });
+
+        m_translationManipulators.InstallPlanarManipulatorMouseUpCallback(
+            [this](const AzToolsFramework::PlanarManipulator::Action& action)
+            {
+                FinishEditing(action.m_start.m_localPosition, action.m_current.m_localOffset);
+            });
+
+        m_translationManipulators.InstallSurfaceManipulatorMouseUpCallback(
+            [this](const AzToolsFramework::SurfaceManipulator::Action& action)
+            {
+                FinishEditing(action.m_start.m_localPosition, action.m_current.m_localOffset);
+            });
+    }
+
+    void RagdollColliderTranslationManipulators::Refresh()
+    {
+        if (!m_ragdollManipulatorData.m_valid || !m_ragdollManipulatorData.m_colliderNodeConfiguration ||
+            m_ragdollManipulatorData.m_colliderNodeConfiguration->m_shapes.empty())
+        {
+            return;
+        }
+
+        m_translationManipulators.SetLocalPosition(m_ragdollManipulatorData.m_colliderNodeConfiguration->m_shapes[0].first->m_position);
+    }
+
+    void RagdollColliderTranslationManipulators::Teardown()
+    {
+        m_translationManipulators.Unregister();
+    }
+
+    void RagdollColliderTranslationManipulators::ResetValues()
+    {
+        m_ragdollManipulatorData.m_colliderNodeConfiguration->m_shapes[0].first->m_position = AZ::Vector3::CreateZero();
+        m_translationManipulators.SetLocalPosition(AZ::Vector3::CreateZero());
+    }
+
+    AZ::Vector3 RagdollColliderTranslationManipulators::GetPosition(const AZ::Vector3& startPosition, const AZ::Vector3& offset)
+    {
+        const float scale = AZ::GetMax(AZ::MinTransformScale, m_ragdollManipulatorData.m_nodeWorldTransform.GetUniformScale());
+        return startPosition + offset / scale;
+    }
+
+    void RagdollColliderTranslationManipulators::OnManipulatorMoved(const AZ::Vector3& startPosition, const AZ::Vector3& offset)
+    {
+        AZ::Vector3 newPosition = GetPosition(startPosition, offset);
+        m_ragdollManipulatorData.m_colliderNodeConfiguration->m_shapes[0].first->m_position = newPosition;
```

Comment:
should this check that `m_shapes` is not empty like in the other functions?

Reply:
Yeah, will add.